src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/DependencyCollection{T}.cs

```diff
@@ -62,22 +77,60 @@ public IEnumerator<T> GetEnumerator()
             {
                 if (!Deletions.Contains(item))
                 {
-                    yield return item;
+                    yield return item.Item;
                 }
             }
 
             foreach (var additions in Additions)
             {
-                yield return additions;
+                yield return additions.Item;
             }
         }
 
         IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
 
+        [Obsolete("Use the newer Add API", error: false)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        bool IDependencyCollection<T>.Add(T item, BuildBreakRisk risk)
+        {
+            var operation = new Operation<T>(item, new() { Risk = risk });
+            if (Contains(item))
```

Comment:
This is duplicating already existing logic. Can't you forward on the call by `Add(item, new() {Risk = risk})`?

Reply:
I tried that, it was not able to resolve the other Add method.